app/src/lib/stores/app-store.ts

```diff
@@ -889,9 +889,29 @@ export class AppStore extends TypedBaseStore<IAppState> {
   }
 
   /** This shouldn't be called directly. See `Dispatcher`. */
-  public _loadNextHistoryBatch(repository: Repository): Promise<void> {
+  public async _loadNextHistoryBatch(repository: Repository): Promise<void> {
     const gitStore = this.getGitStore(repository)
-    return gitStore.loadNextHistoryBatch()
+
+    if (enableCompareSidebar()) {
+      const state = this.getRepositoryState(repository)
+      const { formState } = state.compareState
+      if (formState.kind === ComparisonView.None) {
+        const commits = state.compareState.commitSHAs
+        const lastCommitSha = commits[commits.length - 1]
+
+        const newCommits = await gitStore.loadCommitBatch(lastCommitSha)
+        if (newCommits == null) {
+          return
+        }
+
+        this.updateCompareState(repository, state => ({
+          commitSHAs: commits.concat(newCommits),
```

Comment:
This looks like a good opportunity to use that fancy spread operator`[...commits, ...newCommits]`.

Reply:
I had that initially but avoided it here because I felt the spreading and repopulating the array would be inefficient when dealing with 100 new commits and N x 100 existing commits.

A [quick benchmark with 2 arrays of 20 numbers](https://www.measurethat.net/Benchmarks/Show/3306/0/arrayprototypeconcat-vs-spread-operator-with-more-numbe) on my machine running Chrome shows spread is slightly slower (with a bit more variation in the timings):

<img width="788" src="https://user-images.githubusercontent.com/359239/41737775-992d9984-7566-11e8-9c6c-4e93a1fbdb2c.png">